apps/coa/behaviours/save-document.js

```diff
@@ -13,55 +13,83 @@ module.exports = (documentCategory, name) => superclass => class extends supercl
   }
 
   validateField(key, req) {
-    if (req.body['upload-file']) {
-      const fileUpload = req.files[name];
-
-      if (fileUpload) {
-        const uploadSize = fileUpload.size;
-        const mimetype = fileUpload.mimetype;
-        const uploadSizeTooBig = uploadSize > config.upload.maxFileSizeInBytes;
-        const uploadSizeBeyondServerLimits = uploadSize === null;
-        const invalidMimetype = !config.upload.allowedMimeTypes.includes(mimetype);
-        const invalidSize = uploadSizeTooBig || uploadSizeBeyondServerLimits;
-
-        if (invalidSize || invalidMimetype) {
-          return new this.ValidationError('document-file', {
-            type: invalidSize ? 'maxFileSize' : 'fileType',
-            redirect: undefined
-          });
-        }
-      } else {
-        return new this.ValidationError('document-file', {
-          type: 'required',
-          redirect: undefined
-        });
+    const fileToBeValidated = req.files[name];
+    const documentsByCategory = req.sessionModel.get(documentCategory) || [];
+    const validationErrorFunc = type => new this.ValidationError(key, { type: type });
+
+    // When click Continue without uploading a file.
+    if (req.body.Continue && documentsByCategory.length === 0) {
+      return validationErrorFunc('required');
+    // This Else If statement can be removed once the new design implemented for ID docs Upload screen
+    } else if (!req.body.Continue && !fileToBeValidated) {
+      return validationErrorFunc('required');
+    } else if (fileToBeValidated) {
+      const uploadSize = fileToBeValidated.size;
+      const mimetype = fileToBeValidated.mimetype;
+      const numberOfDocsUploaded = documentsByCategory.length;
+
+      const uploadSizeTooBig = uploadSize > config.upload.maxFileSizeInBytes;
+      const uploadSizeBeyondServerLimits = uploadSize === null;
+
+      const invalidSize = uploadSizeTooBig || uploadSizeBeyondServerLimits;
+      const invalidMimetype = !config.upload.allowedMimeTypes.includes(mimetype);
+
+      const maxIdDocsUploadsExceeds = (documentCategory === 'identity-documents')
+        && numberOfDocsUploaded > config.upload.maxIdDocsUploads;
```

Comment:
`numberOfDocsUploaded > config.upload.maxIdDocsUploads`- This logic needs to be changed like `numberOfDocsUploaded >= config.upload.maxIdDocsUploads` when this new design is added for ID documents upload.

Reply:
changes done